src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,138 @@
+package duke;
+
+import duke.Deadline;
+import duke.Event;
+import duke.Task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+public class TaskList {
+
+    public ArrayList<Task> myList;
+
+    /**
+     * initialises the TaskList (supposed to have loading functionality, but will add on later)
+     */
+
+    public TaskList(ArrayList<Task> parsedList) {
+        myList = parsedList;
+    }
+
+    /**
+     * calls an appropriate method based on the input param
+     * @param input
+     */
+
+    public void action(String input) {
+        if (input.equals("list")) {
+            listTasks();
+        } else if (input.startsWith("done")) {
+            markAsDone(input);
+        } else if (input.startsWith("delete")) {
+            deleteTask(input);
+        } else if (input.startsWith("find")) {
+            findTask(input);
+        } else {
+            addTask(input);
+        }
+    }
+
+    /**
+     * forms a String of all the tasks, for saving into file
+     * @return String
+     */
+    
+    public String output() {
+        String output = "";
+        for (int i = 0; i < myList.size(); i++) {
+            output = output + myList.get(i) + "\n";
+        }
+        return output;
+    }
+
+    private void listTasks() {
+        System.out.println("Here is your to-do list!");
+        for (int i = 0; i < myList.size(); i++) {
+            System.out.println(i + 1 + ". " + myList.get(i).toString());
+        }
+    }
+
+    private void addTask(String input) {
+        try {
+            if (input.startsWith("todo")) {
+                String remaining = input.substring(5);
+                myList.add(new ToDo(remaining));
+            } else if (input.startsWith("deadline")) {
+                String remaining = input.substring(9);
+                String[] segments = remaining.split(" /");
+                LocalDate myDate = LocalDate.parse(segments[1]);
+                myList.add(new Deadline(segments[0], myDate));
+            } else if (input.startsWith("event")) {
+                String remaining = input.substring(6);
+                String[] segments = remaining.split(" /");
+                LocalDate myDate = LocalDate.parse(segments[1]);
+                myList.add(new Event(segments[0], myDate));
+            } else {
+                System.out.println("Invalid task. Please specify the type of task.");
+                return;
+            }
+            System.out.println("I've added this task:");
+            System.out.println(myList.get(myList.size() - 1));
+            System.out.println("Now you have " + myList.size() + " tasks.");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Invalid format. Please follow this format. <type of task> <description> /<date if necessary>");
```

Comment:
Perhaps you could throw a DukeException (or create an exceptions class) and catch it in Duke.java's run function instead of printing a different output for every case?